api/src/game/game.controller.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Body,
+  Controller,
+  Get,
+  HttpException,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { AppGateway } from '../app.gateway';
+import { GameState, Player } from '../types';
+import { GameService } from './game.service';
+import { INITIAL_BOARD } from '../constants';
+
+@Controller('game')
+export class GameController {
+  constructor(
+    private readonly gameService: GameService,
+    private gatewayService: AppGateway,
+  ) {}
+
+  @Post('start')
+  async createGame(): Promise<GameState> {
+    const res = await this.gameService.createGame();
+    return this.gameService.deserializer(res);
+  }
+
+  @Get(':id')
+  async getGameState(@Param('id') id: string): Promise<GameState> {
```

Comment:
your service is named game and you are dealing with gamestate so either rename the service GameStateService/Controller... or change your type to Game as a game is already a state of the game

Reply:
service, module and controller have been renamed into gameState